ocaml/xenopsd/lib/xenops_server.ml

```diff
@@ -1668,9 +1668,12 @@ let rec perform_atomic ~progress_callback ?subtask:_ ?result (op : atomic)
                 in
                 Some e
             | Task.Pending _ ->
-                error "Parallel: queue_atomics_and_wait returned a pending task" ;
+                let err_msg =
+                  Printf.sprintf "Timed out while waiting on task %s" id
```

Comment:
Yesterday we found that `event_wait` inside `queue_atomics_and_wait` returns a Boolean that indicates whether the wait timed out or returned normally, and that this is ignored everywhere. I think that we need to inspect this to know whether this pending task was cancelled or some operation had timed out.

Reply:
Now the code makes a distinction between both when reporting the error.